test/Dapr.E2E.Test.App/Actors/ReentrantActor.cs

```diff
@@ -0,0 +1,57 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Dapr.Actors.Runtime;
+
+namespace Dapr.E2E.Test.Actors.Reentrancy
+{
+    public class ReentrantActor : Actor, IReentrantActor
+    {
+        public ReentrantActor(ActorHost host)
+            : base(host)
+        {
+        }
+
+        public Task Ping()
+        {
+            return Task.CompletedTask;
+        }
+
+        public async Task ReentrantCall(ReentrantCallOptions callOptions)
```

Comment:
Can you add a description in comments (somewhere, doesn't have to be here), of what this test is testing and why it's meaningful?